src/compiler.ts

```diff
@@ -5187,6 +5187,21 @@ export class Compiler extends DiagnosticEmitter {
       }
       case TypeKind.I32:
       case TypeKind.U32: {
+        if (this.options.willOptimize) {
+          // Precompute power if LHS and RHS constants
+          // TODO: move this optimization to AIR
+          if (
+            getExpressionId(leftExpr) == ExpressionId.Const &&
+            getExpressionId(rightExpr) == ExpressionId.Const
+          ) {
+            let leftValue = getConstValueI32(leftExpr);
+            let rightValue = getConstValueI32(rightExpr);
+            return module.i32(i64_low(i64_pow(
```

Comment:
Below, the call to `makeCallDirect` sets `this.currentType`, so I think here (and at the places below) it needs to be set manually.

```suggestion
            this.currentType = type;
            return module.i32(i64_low(i64_pow(
```